File: scripts/patch-test-fixtures.ts
#!/usr/bin/env node
// Add missing new params to all test fixture objects
import { readFileSync, writeFileSync, readdirSync } from 'fs';
import { resolve, dirname } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));

const TEST_DIR = resolve(__dirname, '../src/model/__tests__');
const ROOT_TEST = resolve(__dirname, '../src/model/params.test.ts');
const FILES = [
  ...readdirSync(TEST_DIR).filter(f => f.endsWith('.test.ts')).map(f => resolve(TEST_DIR, f)),
  ROOT_TEST,
];

const NEW_PARAMS = `
    bau_ersatzneubau_effizienz: 1,
    markt_mietbelastungs_grenze: 1,`;

for (const file of FILES) {
  let content = readFileSync(file, 'utf8');
  const hasNewParams = content.includes('bau_ersatzneubau_effizienz');
  if (hasNewParams) continue;

  // Inject after the last param in FULL_PARAMS-like blocks
  // Find the param that's before the closing of the params object
  let modified = false;
  
  // Pattern: find 'infra_wirtschaftsansiedlung: X' followed by newlines and closing brace/blank
  content = content.replace(
    /(\n  infra_wirtschaftsansiedlung: \d+)(\s*\n\s*\})/g,
    (match, line, rest) => {
      modified = true;
      return `${line},${NEW_PARAMS}${rest}`;
    }
  );

  if (modified) {
    writeFileSync(file, content);
    console.log(`Patched: ${file}`);
  }
}

console.log('Done.');

File: scripts/patch-test-fixtures.py
#!/usr/bin/env python3
import os

test_dir = '/Volumes/projects/wohnkosten-simulator/src/model/__tests__'
root_test = '/Volumes/projects/wohnkosten-simulator/src/model/params.test.ts'
files = [os.path.join(test_dir, f) for f in os.listdir(test_dir) if f.endswith('.test.ts')]
files.append(root_test)

NEW_PARAMS = '\n    bau_ersatzneubau_effizienz: 1,\n    markt_mietbelastungs_grenze: 1,'

for filepath in files:
    with open(filepath) as f:
        content = f.read()
    
    if 'bau_ersatzneubau_effizienz' in content:
        print(f'Skip (already patched): {os.path.basename(filepath)}')
        continue
    
    # The params object ends with: infra_wirtschaftsansiedlung: N,\n}
    # Find the last occurrence of this pattern
    idx = content.rfind('infra_wirtschaftsansiedlung: ')
    if idx == -1:
        print(f'No match: {os.path.basename(filepath)}')
        continue
    
    # Find the comma after the number
    comma_idx = content.find(',', idx)
    if comma_idx == -1:
        print(f'No comma after param: {os.path.basename(filepath)}')
        continue
    
    # Find the closing brace
    brace_idx = content.find('}', comma_idx)
    if brace_idx == -1:
        print(f'No closing brace: {os.path.basename(filepath)}')
        continue
    
    new_content = content[:comma_idx+1] + NEW_PARAMS + content[brace_idx:]
    with open(filepath, 'w') as f:
        f.write(new_content)
    print(f'Patched: {os.path.basename(filepath)}')

print('Done.')